servicedispatcher.go

```diff
@@ -162,11 +165,13 @@ func (disp *serviceDispatcher) handleEvent(event upcallEvent) {
 
 // Must be called exactly once to shut down the dispatcher.
```

Comment:
From the comment, I think the bug is actually this method somehow get called more than once. But I think the workaround in this PR is fine for now.